File: app/src/test/java/com/waifusims/wanicchou/SanseidoSearchTest.java
package com.waifusims.wanicchou;

import junit.framework.Assert;

import org.junit.Test;

import data.vocab.jp.JapaneseDictionaryType;
import data.vocab.jp.search.sanseido.SanseidoMatchType;
import data.vocab.jp.search.sanseido.SanseidoSearch;

import static junit.framework.Assert.assertEquals;

/**
 * Test for Sanseido Search
 * Also somewhat indirectly Japanese Vocabulary
 */
public class SanseidoSearchTest {

    @Test
    public void testJJExact() throws Exception {
        SanseidoSearch sanseidoSearch =
                new SanseidoSearch("アニメ", JapaneseDictionaryType.JJ, SanseidoMatchType.EXACT);
        assertEquals(sanseidoSearch.getVocabulary().getWord(), "アニメ");
        assertEquals(sanseidoSearch.getVocabulary().getDictionaryType(), JapaneseDictionaryType.JJ);
        assertEquals(sanseidoSearch.getVocabulary().getDefinition(), "アニメーションの略．");
        assertEquals(sanseidoSearch.getVocabulary().getPitch(), "1");
        assertEquals(sanseidoSearch.getRelatedWords().size(), 1);
    }

    @Test
    public void testJEForwards() throws Exception {
        SanseidoSearch sanseidoSearch =
                new SanseidoSearch("雪", JapaneseDictionaryType.JE, SanseidoMatchType.FORWARDS);
        assertEquals(sanseidoSearch.getVocabulary().getWord(), "雪害");
        assertEquals(sanseidoSearch.getVocabulary().getDictionaryType(), JapaneseDictionaryType.JE);
        assertEquals(sanseidoSearch.getVocabulary().getDefinition(), "snow damage．");
        assertEquals(sanseidoSearch.getVocabulary().getPitch(), "");
        int size = sanseidoSearch.getRelatedWords().size();
        assertEquals(size, 20);
    }

    @Test
    public void testEmptyWord() throws Exception {
        try{
            SanseidoSearch sanseidoSearch =
                    new SanseidoSearch("", JapaneseDictionaryType.EJ, SanseidoMatchType.BACKWARDS);
            Assert.fail("Should have thrown IllegalArgumentException.");
        }
        catch (IllegalArgumentException e){

        }

    }

    @Test
    public void testNullDictionaryType() throws Exception {
        try{
            SanseidoSearch sanseidoSearch =
                    new SanseidoSearch("テスト", null, SanseidoMatchType.EXACT);
            Assert.fail("Should have thrown IllegalArgumentException.");
        }
        catch (IllegalArgumentException e){ }
    }

    @Test
    public void testNullMatchType() throws Exception {
        try{
            SanseidoSearch sanseidoSearch =
                    new SanseidoSearch("テスト", JapaneseDictionaryType.JJ, null);
            Assert.fail("Should have thrown IllegalArgumentException.");
        }
        catch (IllegalArgumentException e){ }
    }
}

File: app/src/test/java/com/limegrass/wanicchou/SanseidoSearchTest.kt
package com.limegrass.wanicchou

/**
 * Test for Sanseidou Search
 * Also somewhat indirectly Japanese DictionaryEntry
 */
//class SanseidoSearchTest {
//
//    @Test
//    @Throws(Exception::class)
//    fun testJJExact() {
//        val sanseidoSearch = SanseidoSearch("アニメ", JapaneseDictionaryType.JJ, SanseidouMatchType.EXACT)
//        assertEquals(sanseidoSearch.dictionaryEntry!!.word, "アニメ")
//        assertEquals(sanseidoSearch.dictionaryEntry!!.dictionaryType, JapaneseDictionaryType.JJ)
//        assertEquals(sanseidoSearch.dictionaryEntry!!.definition, "アニメーションの略．")
//        assertEquals(sanseidoSearch.dictionaryEntry!!.pitch, "1")
//        assertEquals(sanseidoSearch.relatedWords!!.size, 1)
//    }
//
//    @Test
//    @Throws(Exception::class)
//    fun testJEForwards() {
//        val sanseidoSearch = SanseidoSearch("雪", JapaneseDictionaryType.JE, SanseidouMatchType.FORWARDS)
//        assertEquals(sanseidoSearch.dictionaryEntry!!.word, "雪害")
//        assertEquals(sanseidoSearch.dictionaryEntry!!.dictionaryType, JapaneseDictionaryType.JE)
//        assertEquals(sanseidoSearch.dictionaryEntry!!.definition, "snow damage．")
//        assertEquals(sanseidoSearch.dictionaryEntry!!.pitch, "")
//        val size = sanseidoSearch.relatedWords!!.size
//        assertEquals(size, 20)
//    }
//
//    @Test
//    @Throws(Exception::class)
//    fun testEmptyWord() {
//        try {
//            val sanseidoSearch = SanseidoSearch("", JapaneseDictionaryType.EJ, SanseidouMatchType.BACKWARDS)
//            Assert.fail("Should have thrown IllegalArgumentException.")
//        } catch (e: IllegalArgumentException) {
//
//        }
//
//    }
//
//    @Test
//    @Throws(Exception::class)
//    fun testNullDictionaryType() {
//        try {
//            val sanseidoSearch = SanseidoSearch("テスト", null, SanseidouMatchType.EXACT)
//            Assert.fail("Should have thrown IllegalArgumentException.")
//        } catch (e: IllegalArgumentException) {
//        }
//
//    }
//
//    @Test
//    @Throws(Exception::class)
//    fun testNullMatchType() {
//        try {
//            val sanseidoSearch = SanseidoSearch("テスト", JapaneseDictionaryType.JJ, null)
//            Assert.fail("Should have thrown IllegalArgumentException.")
//        } catch (e: IllegalArgumentException) {
//        }
//
//    }
//}